server/chat-plugins/scavenger-games.ts

```diff
@@ -7,29 +7,49 @@
  * @license MIT license
  */
 
-'use strict';
+import {ScavengerHunt} from './scavengers';
+
+export type TwistEvent = (
+	this: ScavengerHunt,
+	...args: any[]
+) => void;
+interface Twist {
+	name: string;
+	id: string;
+	desc?: string;
+	[eventid: string]: string | number | TwistEvent | undefined;
+}
+interface GameMode {
+	name: string;
+	id: string;
+	mod: Twist;
+	round?: number;
+	leaderboard?: true;
+	[k: string]: any;
+}
 
 class Leaderboard {
-	constructor(game) {
-		this.game = game;
+	data: AnyObject;
+
+	constructor() {
 		this.data = {};
 	}
 
-	addPoints(name, aspect, points, noUpdate) {
-		const userid = toID(name);
+	addPoints(name: string, aspect: string, points: number, noUpdate?: boolean) {
+		const userid: string = toID(name);
 
 		if (!userid || userid === 'constructor' || !points) return this;
 		if (!this.data[userid]) this.data[userid] = {name: name};
 
 		if (!this.data[userid][aspect]) this.data[userid][aspect] = 0;
-		this.data[userid][aspect] += points;
+		this.data[userid][aspect] = this.data[userid][aspect] = points;
```

Comment:
This seems like it would do something completely different from the earlier code (and also looks like it's clearly a bug). What was your intent here? Did the old code not work?

Reply:
That was a mistake - I dont even know how that happened.  I'm not enough of a furbrain to do something like this.